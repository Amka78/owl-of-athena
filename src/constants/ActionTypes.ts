const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
