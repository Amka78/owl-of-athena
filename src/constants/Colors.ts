const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
