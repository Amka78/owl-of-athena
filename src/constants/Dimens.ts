export default {
    toolbar_title_text_size: 14,
    toolbar_button_text_size: 14,
    toolbar_button_margin_horizontal: 8,

    view_padding_bottom: 12,

    content_title_text_size: 24,
    content_title_margin_top: 32,
    content_title_margin_bottom: 48,
    content_text_size: 16,
    content_margin_horizontal: 48,

    snackbar_text_size: 8,

    statusbar_height: 8,

    navbar_height: 64,

    tip_text_size: 13,
    tip_text_margin_bottom: 13,

    error_text_size: 13,
    error_text_margin_bottom: 18,

    button_text_size: 24,
    button_height: 50,
    button_margin_bottom: 16,
    button_radius: 16,
    button_flat_text_size: 14,
    button_flat_margin_bottom: 12,
    button_flat_height: 28,

    input_text_size: 12,
    input_margin_bottom: 16,

    checkbox_text_size: 12,
    checkbox_outer_diameter: 24,
    checkbox_inner_diameter: 16,
    checkbox_gap: 4,
    checkbox_offset: 2,
    checkbox_padding_left: 16,
    checkbox_padding_right: 24,
    checkbox_margin_top: 16,
    checkbox_margin_bottom: 8,

    option_text_size: 14,
    option_height: 20,
    option_margin_bottom: 12,

    time_picker_height: 150,
    time_picker_scale: 1,
    time_picker_padding_horizontal: 0,

    circular_progress_diameter: 96,
    circular_progress_text_size: 32,

    // home fragment dimensions
    home_alarm_time_text_size: 48,
    home_alarm_meridian_text_size: 22,

    // session fragment dimensions
    session_label_text_size: 20,
    session_time_text_size: 28,
    session_sleep_score_text_size: 48,
    session_notes_text_size: 20,

    // session list fragment dimensions
    session_list_item_padding: 12,

    session_list_section_height: 24,
    session_list_section_padding_left: 24,
    session_list_section_text_size: 16,

    session_list_item_title_text_size: 20,
    session_list_item_subtitle_text_size: 15,
    session_list_item_text_margin_left: 18
};
