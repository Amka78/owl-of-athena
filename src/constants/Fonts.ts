export default {
    primaryRegular: "calibre_app_regular",
    primarySemiBold: "calibre_app_semibold"
};
