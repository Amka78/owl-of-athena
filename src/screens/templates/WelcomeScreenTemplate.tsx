//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import {
    Button,
    ButtonProps,
    ContentText,
    ContentTextProps,
    ContentTitle,
    ContentTitleProps,
    StandardView,
} from "../../components";
//#endregion

//#region Types
export type WelcomeScreenTemplateProps = {
    contentTitle: ContentTitleProps;
    contentText: ContentTextProps;
    standaloneButton: ButtonProps;
    loginButton: ButtonProps;
    signupButton: ButtonProps;
};
//#endregion

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    return (
        <StandardView>
            <ContentTitle {...props.contentTitle}></ContentTitle>
            <ContentText {...props.contentText}></ContentText>
            <View>
                <Button {...props.standaloneButton}></Button>
                <Button {...props.loginButton}></Button>
                <Button {...props.signupButton}></Button>
            </View>
        </StandardView>
    );
};
//#endregion
