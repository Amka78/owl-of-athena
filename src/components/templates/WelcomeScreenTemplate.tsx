//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import { Message, MessageKeys } from "../../constants";
import { useLocale } from "../../hooks";
import { Button, ContentText, ContentTitle, LeftSideButton } from "../atoms";
import { InternalView, RightSideButton } from "../molecules";
//#endregion

//#region Types
export type WelcomeScreenTemplateProps = {
    onStandalonePress: () => void;
    onLoginPress: () => void;
    onSignupPress: () => void;
    dimens: { isLargeWidth: boolean; isHorizontal: boolean };
    locale?: string;
};
//#endregion

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    useLocale(props.locale);

    const isTwoColumnsBottoms =
        props.dimens.isHorizontal || props.dimens.isLargeWidth;
    return (
        <InternalView>
            <ContentTitle>
                {Message.get(MessageKeys.welcome_title)}
            </ContentTitle>
            <ContentText>{Message.get(MessageKeys.welcome_text)}</ContentText>
            <View style={{ alignItems: "center" }}>
                <View
                    style={{
                        flexDirection: isTwoColumnsBottoms ? "row" : "column",
                    }}
                >
                    <LeftSideButton
                        onPress={props.onLoginPress}
                        isLargeWidth={isTwoColumnsBottoms}
                    >
                        {Message.get(MessageKeys.welcome_login_button)}
                    </LeftSideButton>
                    <RightSideButton
                        onPress={props.onSignupPress}
                        isLargeWidth={isTwoColumnsBottoms}
                    >
                        {Message.get(MessageKeys.welcome_signup_button)}
                    </RightSideButton>
                </View>
                <Button onPress={props.onStandalonePress}>
                    {Message.get(MessageKeys.welcome_standalone_button)}
                </Button>
            </View>
        </InternalView>
    );
};
//#endregion
