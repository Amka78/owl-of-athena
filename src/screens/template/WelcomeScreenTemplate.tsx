//#region Import Modules
import React, { FunctionComponent } from "react";
import { View } from "react-native";

import {
    Button,
    ContentText,
    ContentTitle,
    StandardView,
} from "../../components";
import { MessageLocalizationParam } from "../../constants/Message";
//#endregion

export type WelcomeScreenTemplateProps = {
    contentTitleText: MessageLocalizationParam;
    contentText: MessageLocalizationParam;
    standaloneButtonText: MessageLocalizationParam;
    standaloneButtonPress: () => void;
    loginButtonText: MessageLocalizationParam;
    loginButtonPress: () => void;
    signupButtonText: MessageLocalizationParam;
    signupButtonPress: () => void;
};

//#region Component
export const WelcomeScreenTemplate: FunctionComponent<WelcomeScreenTemplateProps> = (
    props: WelcomeScreenTemplateProps
) => {
    return (
        <StandardView>
            <ContentTitle>{props.contentTitleText}</ContentTitle>
            <ContentText>{props.contentText}</ContentText>
            <View>
                <Button onPress={props.standaloneButtonPress}>
                    {props.standaloneButtonText}
                </Button>
                <Button onPress={props.loginButtonPress}>
                    {props.loginButtonText}
                </Button>
                <Button onPress={props.signupButtonPress}>
                    {props.signupButtonText}
                </Button>
            </View>
        </StandardView>
    );
};
//#endregion
