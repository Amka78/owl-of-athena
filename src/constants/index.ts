import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
