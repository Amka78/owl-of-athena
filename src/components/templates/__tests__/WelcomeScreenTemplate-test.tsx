import "jest-enzyme";
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";
import { WelcomeScreenTemplate } from "../WelcomeScreenTemplate";
import {
    createMock,
    DesktopDimension,
    toJson,
} from "../../../utils/TestHelper";
import { Message } from "../../../constants";

let component: ShallowWrapper<unknown, unknown, unknown>;

describe("WelcomeScreenTemplate UnitTest", () => {
    it.each(["us", "ja"])("renders correctly", (locale: string) => {
        const testEvent: () => void = () => {
            return;
        };
        Message.setLocale(locale);
        component = createMock(
            <WelcomeScreenTemplate
                onLoginPress={testEvent}
                onSignupPress={testEvent}
                onStandalonePress={testEvent}
                dimens={DesktopDimension}
            ></WelcomeScreenTemplate>
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
