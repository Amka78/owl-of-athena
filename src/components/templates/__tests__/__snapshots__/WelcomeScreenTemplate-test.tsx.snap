// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreenTemplate UnitTest renders correctly 1`] = `
<Component>
  <Component>
    Welcome to Aurora!
  </Component>
  <Component>
    Aurora’s advanced software will help you improve dream recall, achieve sounder sleep,and accelerate your journey towards maximizing your potential. Sweet dreams!
  </Component>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Component
        isLargeWidth={true}
        onPress={[Function]}
      >
        Log In
      </Component>
      <Component
        isLargeWidth={true}
        onPress={[Function]}
      >
        Sign Up
      </Component>
    </View>
    <Component
      onPress={[Function]}
    >
      StandAlone
    </Component>
  </View>
</Component>
`;

exports[`WelcomeScreenTemplate UnitTest renders correctly 2`] = `
<Component>
  <Component>
    Auroraにようこそ！
  </Component>
  <Component>
    このソフトウェアは貴方のドリームリコールを向上させます。より良い睡眠を実現し、明晰夢を体験する可能性があがります。 良い夢を!
  </Component>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Component
        isLargeWidth={true}
        onPress={[Function]}
      >
        ログイン
      </Component>
      <Component
        isLargeWidth={true}
        onPress={[Function]}
      >
        サインアップ
      </Component>
    </View>
    <Component
      onPress={[Function]}
    >
      スタンドアローン
    </Component>
  </View>
</Component>
`;
